### PR TITLE
Removing certificate pinning code and references

### PR DESCRIPTION
**Pull request by eli-iser**
Removing certificate pinning code as we no longer want the SDK to enforce the certificate pinning. The README will be updated on how this should be achieved by the SDK users if desired.
This also removes the references to the cert pinning function from the example projects that had it.

## **Checklist**

**General Contributing**
- [ ] Have you read the Code of Conduct and signed the [CLA](https://opensource.dropbox.com/cla/)?

**Is This a Code Change?**
- [ ] Non-code related change (markdown/git settings etc)
- [X] SDK Code Change
- [X] Example/Test Code Change

**Validation**
- [X] Does this code build successfully?
- [X] Do all tests pass?
- [X] Does Stylecop pass?

### CODE REVIEW

**Review by sahujapersonal — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -35,6 +35,11 @@ We provide [Examples][examples] to help get you started with a lot of the basic
     - [Simple Business Dashboard](https://github.com/dropbox/dropbox-sdk-dotnet/tree/main/dropbox-sdk-dotnet/Examples/SimpleBusinessDashboard) - This is a demo of a business dashboard
     - [Universal Demo](https://github.com/dropbox/dropbox-sdk-dotnet/tree/main/dropbox-sdk-dotnet/Examples/UniversalDemo/UniversalDemo) - This is an example of how to use the SDK across multiple platforms
 
+### Certificate pinning
+
+If certificate pinning is desired when using the Dropbox SDK to validate that the right Dropbox servers are being hit, see [Microsoft's docuemntation](https://learn.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.servercertificatevalidationcallback?view=net-8.0) on how to perform this validation.
+The Dropbox certificates can be found here.
```
Comment (maxbelanger) [inline README.md:41]: This is missing a link, is that still TBD?

Comment (eli-iser) [reply README.md:41]: Correct. AFAIK we still don't have the location of the public facing certificates.

**Review by maxbelanger — APPROVED** [review]
lgtm, pending adding the right link. cc @april can you take a look at the messaging?

File: README.md (hunk 1 above)
Comment (greg-db) [reply README.md:41]: Yea, the last I recall, we don't have the if/when/where/how of this sorted out yet, so we should probably just take this out for now.

Comment (eli-iser) [reply README.md]: Removed this doc block until we figured out our comms

**Pull request merged**